### PR TITLE
Enable `thresh_min_proba` to be `None`

### LINKED ISSUE #594: MESMER-X: can `threshold_min_proba` be `None`?
There are two places where we test `if self.threshold_min_proba is None`

https://github.com/MESMER-group/mesmer/blob/980521d8bd2a09495d0bce63036834e35c4db4ca/mesmer/mesmer_x/train_l_distrib_mesmerx.py#L811

https://github.com/MESMER-group/mesmer/blob/980521d8bd2a09495d0bce63036834e35c4db4ca/mesmer/mesmer_x/train_l_distrib_mesmerx.py#L1204

However, there is also a test which fails if it is `None`:

https://github.com/MESMER-group/mesmer/blob/980521d8bd2a09495d0bce63036834e35c4db4ca/mesmer/mesmer_x/train_l_distrib_mesmerx.py#L507

Therefore question @yquilcaille 
- should we remove the `is None` checks?
- or should we fix to allow `is None`?

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**
Before there was check hindering `thresh_min_proba` to be `None` even though this can be useful, for example for the `truncnorm`, see the issue below.

 - [x] Closes #594 
 - [x] Tests added
 - [x] Fully documented, including `CHANGELOG.rst`

### CODE REVIEW

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 1]
```diff
@@ -526,8 +528,10 @@ def __init__(
         self.data_targ_addtest = data_targ_addtest
         self.data_preds_addtest = data_preds_addtest
 
-        if (threshold_min_proba <= 0) or (1 < threshold_min_proba):
-            raise ValueError("`threshold_min_proba` must be in [0;1[")
+        if threshold_min_proba is not None and (
+            (threshold_min_proba <= 0) or (0.5 <= threshold_min_proba)
+        ):
+            raise ValueError("`threshold_min_proba` must be in (0,0.5)")
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:534]:
```suggestion
            raise ValueError("`threshold_min_proba` must be in (0, 0.5)")
```

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 2]
```diff
@@ -341,10 +341,12 @@ def __init__(
             data, but will test that this data remains valid. Important to avoid points
             out of support of the distribution.
 
-        threshold_min_proba : float, default: 1e-9
-            Will test that each point of the sample and added sample have their
-            probability higher than this value. Important to ensure that all points are
-            feasible with the fitted distribution.
+        threshold_min_proba : float or None, default: 1e-9
+            if not None imposes a check during the fitting that every sample fulfills
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:345]:
```suggestion
            If numeric imposes a check during the fitting that every sample fulfills
```

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 3]
```diff
@@ -341,10 +341,12 @@ def __init__(
...
+            `cdf(sample) >= threshold_min_proba and 1-cdf(sample) >= threshold_min_proba`,
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:346]:
```suggestion
            `cdf(sample) >= threshold_min_proba and 1 - cdf(sample) >= threshold_min_proba`,
```

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 4]
```diff
@@ -341,10 +341,12 @@ def __init__(
...
+            i.e. each sample lies within some confidence interval of the distribution.
+            Note that it follows that threshold_min_proba math::\\in (0,0.5). Important to
+            ensure that all points are feasible with the fitted distribution.
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:349]:
```suggestion
            ensure that all points are feasible with the fitted distribution.
            If `None` this test is skipped.
```

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 5]
```diff
@@ -789,15 +793,17 @@ def _test_evol_params(self, distrib, data):
         return True
 
     def _test_proba_value(self, distrib, data):
-        # tested values must have a minimum probability of occurring, i.e. be in a
-        # confidence interval
+        """
+        Test that all cdf(data) >= threshold_min_proba and 1-cdf(data) >= threshold_min_proba
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:797]:
```suggestion
        Test that all cdf(data) >= threshold_min_proba and 1 - cdf(data) >= threshold_min_proba
```

**Review by mathause — APPROVED** [review]
Some small nits otherwise good to go! Thanks.

**Comment by yquilcaille** [comment]
It looks good to me as well :)
